crates/core/src/model/scalar/datetime.rs

```diff
@@ -182,19 +182,52 @@ mod friendly_date_time {
     }
 }
 
+// Parse a string representing the date and time.
+//
+// Accepts any of the formats listed below and builds return an `Option`
+// containing a `DateTimeImpl`.
+//
+// Supported formats:
+//
+// * `default` - `YYYY-MM-DD HH:MM:SS`
+// * `day_month` - `DD Month YYYY HH:MM:SS`
+// * `day_mon` - `DD Mon YYYY HH:MM:SS`
+// * `mdy` -  `MM/DD/YYYY HH:MM:SS`
+// * `dow_mon` - `Dow Mon DD HH:MM:SS YYYY`
+//
+// Offsets in one of the following forms, and are catenated with any of
+// the above formats.
+//
+// * `+HHMM`
+// * `-HHMM`
+//
+// Example:
+//
+// * `dow_mon` format with an offset: "Tue Feb 16 10:00:00 2016 +0100"
 fn parse_date_time(s: &str) -> Option<DateTimeImpl> {
+    use regex::Regex;
+    use time::macros::format_description;
+
     const USER_FORMATS: &[&[time::format_description::FormatItem<'_>]] = &[
-        time::macros::format_description!("[day] [month repr:long] [year] [hour]:[minute]:[second] [offset_hour sign:mandatory][offset_minute]"),
-        time::macros::format_description!("[day] [month repr:short] [year] [hour]:[minute]:[second] [offset_hour sign:mandatory][offset_minute]"),
         DATE_TIME_FORMAT,
+        format_description!("[day] [month repr:long] [year] [hour]:[minute]:[second] [offset_hour sign:mandatory][offset_minute]"),
+        format_description!("[day] [month repr:short] [year] [hour]:[minute]:[second] [offset_hour sign:mandatory][offset_minute]"),
+        format_description!("[month]/[day]/[year] [hour]:[minute]:[second] [offset_hour sign:mandatory][offset_minute]"),
+        format_description!("[weekday repr:short] [month repr:short] [day padding:none] [hour]:[minute]:[second] [year] [offset_hour sign:mandatory][offset_minute]"),
     ];
 
-    match s {
-        "now" => Some(DateTimeImpl::now_utc()),
-        _ => USER_FORMATS
+    if let "" = s {
+        None
+    } else if let "now" | "today" | "Today" = s {
+        Some(DateTimeImpl::now_utc())
+    } else {
+        let offset_re = Regex::new(r"[+-][01][0-9]{3}$").unwrap();
+
+        let s = s.to_owned() + if offset_re.is_match(s) { "" } else { " +0000" };
```

Comment:
Please simplify this by breaking the conditional expression out into its own variable
```rust
let offset = if offset_re.is_match(s) { "" } else { " +0000" };
let s = s.to_owned() + offset;
```

Reply:
Done.